src/ORM/DataObject.php

```diff
@@ -1258,6 +1258,8 @@ public function forceChange()
     public function validate(): ValidationResult
     {
         $result = ValidationResult::create();
+        $result->setModelClass(static::class);
+        $result->setRecordID($this->ID);
```

Comment:
Since these aren't available through the constructor anymore, calling two methods to set a single model feels a bit off. Maybe there should be a `$result->setModel($this);`?
Don't have to implement this, just a thought.

Reply:
I think it makes sense to keep separate as you may want to set a non-numeric recordID that doesn't relate to a DataObject/model.  I don't want to add what is essentially a duplicate method

Reply:
I don't understand how having an ID without knowing what class that ID belongs to would ever be helpful, but probably not worth discussing since the methods are there to set the stuff.